libgdf/src/unaryops.cu

```diff
@@ -80,11 +82,57 @@ struct UnaryOp {
 };
 
 
+template<typename T, typename Tout, typename F>
+struct UnaryOpArrow {
+    static
+    gdf_error launch(gdf_column *input, gdf_column *output) {
+        std::shared_ptr<arrow::PrimitiveArray> input_arrow, output_arrow;
```

Comment:
What is the point of this? All you do is convert the input and output gdf_columns to arrow PrimitiveArrays, but then all you use those arrays for is to get their data pointers, which are the same pointers from the gdf_columns. It seems like a pointless change.